libs/seiscomp/io/recordstream/slconnection.cpp

```diff
@@ -436,7 +436,6 @@ Record *SLConnection::next() {
 					SEISCOMP_DEBUG("Handshaking SeedLink server at %s", _serverloc.c_str());
 					handshake();
 					_readingData = true;
-					_retriesLeft = -1;
```

Comment:
I think the intention was to break the loop if the handshake was successful (after X retries) but the connection failed during streams caused by either a network issue, a server restart or a timeout. If you configured a maximum amount of retries then it only affects the handshake, e.g. to wait until the server is available.

If streaming is aborted for some reason then we should repeat the handshake in any case and if it fails after X retries then it will cancel the recordstream.

Reply:
> if you configured a maximum amount of retries then it only affects the handshake, e.g. to wait until the server is available.

I was surprised by this behaviour and for this reason I debugged it, but If this is the intended behviour than I am happy with that. 

> If streaming is aborted for some reason then we should repeat the handshake in any case

This is exactly how it works after this PR.

> and if it fails after X retries then it will cancel the recordstream.

The total amount of failures is kept and it the sum of all the failuers that has happened either during the handshake or during the streaming